awx/ui-next/Makefile

```diff
@@ -0,0 +1,87 @@
+ui_next_mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+ui_next_dir := $(dir $(ui_next_mkfile_path))
+
+ifeq ($(current_dir), undefined)
+	ui_next_dir := $(dir $(ui_next_mkfile_path))
+else
+	ui_next_dir := $(shell realpath --relative-to $(current_dir) $(ui_next_dir))/
+endif
+
+ANSIBLE_UI_GIT_REPO_HTTPS ?= https://github.com/ansible/ansible-ui.git
+ANSIBLE_UI_GIT_REPO_SSH ?= git@github.com:ansible/ansible-ui.git
+
+ANSIBLE_UI_GIT_BRANCH ?= main
+
+ANSIBLE_UI_LOCAL ?=
+
+ANSIBLE_UI_DIR ?= $(ui_next_dir)ansible-ui
+
+.PHONY: ui-next/clone-https
+## Shallow clone the ui-next repo via https skip if ANSIBLE_UI_GIT_REPO_HTTPS is undefined
+ui-next/clone-https:
+	@if [ -z "$(ANSIBLE_UI_GIT_REPO_HTTPS)" ]; then \
+		echo "SKIP: ui-next/clone-https. ANSIBLE_UI_GIT_REPO_HTTPS is not set."; \
+	elif [ -d "$(ANSIBLE_UI_DIR)" ]; then \
+		echo "SKIP: ui-next/clone-https. $(ANSIBLE_UI_DIR) already exists."; \
+	else \
+		git clone --depth 1 --branch $(ANSIBLE_UI_GIT_BRANCH) $(ANSIBLE_UI_GIT_REPO_HTTPS) $(ANSIBLE_UI_DIR) || true; \
+	fi
+
+.PHONY: ui-next/clone-ssh
+## Shallow clone the ui-next repo via ssh.
+ui-next/clone-ssh:
+	@if [ -z "$(ANSIBLE_UI_GIT_REPO_SSH)" ]; then \
```

Comment:
Ohh. We should have style guides for awx-ish projects, and we could include a GNU Make style guide/section. But, for now I just want to say if you do decide to keep handling these somewhat mutually exclusive vars this way, then at least wrap these repetitive lines in a "user-defined function" (really, a multi-line macro with optional params):
```
# $(call maybe-clone,repo-var)
define maybe-clone
    @if [ -z $$1 ]; then \ echo "skip: not set"; \
        elif [ -d "$ANSIBLE_UI_DIR" ]; then \
            echo 'skip: ui dir exists'; \
        else \
            git clone --depth 1 --branch $(branch) $($1) $(ANSIBLE_UI_DIR) || true; \
       fi
endef```
Something like that. Didn't test.
But, then you could, e.g., `$(call maybe-clone,ANSIBLE_UI_GIT_REPO_SSH)` It could be simplified further by not parameterizing it, too. Just put the text you want in a `define`, and make two of them if you need to. I think even this might be simpler if you collapse the `REPO_{SSH,HTTPS}` gargoyles, too.

Reply:
oh yeah, you might skip the subshells with like,
```
$(if $(wildcard $(dir $(ANSIBLE_UI_DIR)),$(warning "skippity doo da"),git clone --depth 1 ...)
```
My syntax might be off. Getting back into the make headspace here.